File: Problems 3400-3499/3442. Maximum Difference Between Even and Odd Frequency I/solution.cpp
class Solution {
    public:
        int maxDifference(string s) {
            vector<int> freq(26, 0);
    
            for (char c : s) {
                freq[c-'a']++;
            }
    
            int maxOdd = 0;
            int minEven = s.length() + 1;
            for (int count : freq) {
                if (count == 0) continue;
                if ((count % 2) == 0) {
                    minEven = min(minEven, count);
                } else {
                    maxOdd = max(maxOdd, count);
                }
            }
    
            return maxOdd - minEven;
        }
    };

File: Problems 3400-3499/3442. Maximum Difference Between Even and Odd Frequency I/solution.py
class Solution:
    def maxDifference(self, s: str) -> int:
        freq = [0] * 26

        for c in s:
            freq[ord(c) - ord('a')] += 1
        
        n = len(s)
        minEven = n + 1
        maxOdd = 0
        for count in freq:
            if count == 0: continue
            if (count % 2) == 0:
                minEven = min(minEven, count)
            else:
                maxOdd = max(maxOdd, count)
        
        return maxOdd - minEven